main.py

```diff
@@ -558,24 +562,14 @@ def main(cl_arguments):
     if args.do_full_eval:
         log.info("Evaluating...")
         splits_to_write = evaluate.parse_write_preds_arg(args.write_preds)
-        if args.do_target_task_training or (args.load_model and not args.do_pretrain):
-            # If we either do target task training, or if we only evaluate
-            # without pretraining or target task training
-            # then we evaluate on the target tasks.
-            for task in target_tasks:
-                # Find the task-specific best checkpoint to evaluate on.
-                ckpt_path = get_best_checkpoint_path(args, "eval", task.name)
-                assert ckpt_path is not None
-                load_model_state(model, ckpt_path, args.cuda, skip_task_models=[], strict=strict)
-
-                evaluate_and_write(args, model, [task], splits_to_write)
-        elif args.do_pretrain:
```

Comment:
Note that now, if we pretrain on a task and then evaluate on that task, we have to put that task in both pretrain_tasks and target_tasks.

Reply:
Sounds right!